src/index.ts

```diff
@@ -27,7 +30,62 @@ const defaultAppFns: ApplicationFunction[] = [
 // tslint:enable:no-var-requires
 
 export class Probot {
+  public static async run (appFn: ApplicationFunction | string[]) {
+    require('dotenv').config()
+
+    const pkgConf = require('pkg-conf')
+    const program = require('commander')
+
+    const readOptions = (): Options => {
+      if (Array.isArray(appFn)) {
+        program
+          .usage('[options] <apps...>')
+          .option('-p, --port <n>', 'Port to start the server on', process.env.PORT || 3000)
+          .option('-W, --webhook-proxy <url>', 'URL of the webhook proxy service.`', process.env.WEBHOOK_PROXY_URL)
+          .option('-w, --webhook-path <path>', 'URL path which receives webhooks. Ex: `/webhook`', process.env.WEBHOOK_PATH)
+          .option('-a, --app <id>', 'ID of the GitHub App', process.env.APP_ID)
+          .option('-s, --secret <secret>', 'Webhook secret of the GitHub App', process.env.WEBHOOK_SECRET)
+          .option('-P, --private-key <file>', 'Path to certificate of the GitHub App', process.env.PRIVATE_KEY_PATH)
+          .parse(appFn)
+
+        return {
+          cert: findPrivateKey(program.privateKey) || undefined,
+          id: program.app,
+          port: program.port,
+          secret: program.secret,
+          webhookPath: program.webhookPath,
+          webhookProxy: program.webhookProxy
+        }
+      } else {
```

Comment:
You don’t ned the else here because of the return above

Reply:
I just copied the code from `probot-run.js`. But I can change that.